tracks/bash/exercises/raindrops/mentoring.md

```diff
@@ -39,3 +39,80 @@ See the [two-fer notes](https://raw.githubusercontent.com/exercism/website-copy/
         > Within an expression, shell variables may also be referenced by name without using the parameter expansion syntax.
 
       [https://www.gnu.org/software/bash/manual/bash.html#Shell-Arithmetic](https://www.gnu.org/software/bash/manual/bash.html#Shell-Arithmetic)
+
+
+# Bash Guidance
+
+
+## Preflight checks
+
+Check if the interpreter directive, aka hash-pling `#!` is present. Increased portability is achieved by not hard-coding this path. An indirect lookup with `env` is current best practice. Contrast:
+
+```
+#! /bin/bash
+#! /usr/bin/env bash
+```
+
+
+## Interactive prompt self-help
+
+Query if there is an awareness of classic documentation that is typically available:
+
+```
+$ man  bash
+$ info bash
+```
+
+A leaner alternative, [`tldr` (https://github.com/tldr-pages/tldr/)](https://github.com/tldr-pages/tldr/), reduces documentation down to common usage examples:
+
+```
+$ tldr bash
+
+bash
+
+Bourne-Again SHell, an `sh`-compatible command-line interpreter.
+More information: <https://gnu.org/software/bash/>.
+
+- Start an interactive shell session:
+    bash
+
+- Execute a command and then exit:
+    bash -c "command"
+
+...
+
+```
+
+
+## Static analysis & linting
+
+[ShellCheck (https://shellcheck.net/)](https://shellcheck.net/) by Vidar Holen is a stable pre 1.0 tool that provides a set of clear, editable rationales for each type of detected issue. Other less problematic recommendations can also be requested:
+
+```
+$ shellcheck <file>
+$ shellcheck --enable=all <file>
+```
+
+
+## Debug
+
+Consider Aaron Maxwell's [Bash Strict Mode (http://redsymbol.net/articles/unofficial-bash-strict-mode/)](http://redsymbol.net/articles/unofficial-bash-strict-mode/) to help reduce an invisible issue down to a minimal working example:
+
+```
+set -Eeuo pipefail
+IFS=$'\n\t'
+```
+
+
+## Resources
+
+A special mention must be made of Michael Mol's chrestomathy site, [Rosetta Code (https://rosettacode.org/wiki/Category:Bash)](https://rosettacode.org/wiki/Category:Bash). Comparing solutions with other known or even unknown languages can be rather instructive.
+
```

Comment:
https://wiki.bash-hackers.org/scripting/tutoriallist might be worth a mention, as a curated/ranked list of tutorials.

Reply:
I'm adding it.